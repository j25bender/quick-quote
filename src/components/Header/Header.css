@import url("https://fonts.googleapis.com/css?family=Courgette");
header {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  background-color: #4F5D75;
  height: 10vh;
  font-family: 'Courgette', cursive;
  padding-bottom: 10px;
  margin-bottom: 50px; }

.header-section h1 {
  margin: 0; }

a.nav {
  align-self: center;
  text-decoration: none;
  font-size: 20px;
  color: #2D3142; }

a.nav.active {
  align-self: flex-end;
  font-weight: bold;
  color: #FF824C; }
