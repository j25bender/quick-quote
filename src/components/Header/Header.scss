@import url('https://fonts.googleapis.com/css?family=Courgette');
// https://coolors.co/2d3142-4f5d75-bfc0c0-ffffff-ff824c

header {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-items: center;
  background-color: #4F5D75;
  height: 10vh;
  font-family: 'Courgette', cursive;
  padding-bottom: 10px;
}

.header-section h1 {
  margin: 0;
}

a.nav {
  align-self: center;
  text-decoration: none;
  font-size: 20px;
  color: #2D3142;
}

a.nav.active {
  align-self: flex-end;
  font-weight: bold;
  color: #FF824C;
}