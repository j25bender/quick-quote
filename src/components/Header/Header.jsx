import React, { Component } from 'react';
import './Header.css';
import { NavLink } from 'react-router-dom';

export class Header extends Component {
  render() {
    return (
      <div>
        <div className="header-section">
          <header>
          <h1>QUICK QUOTE</h1>
            <NavLink to='/home' className='nav'>Home</NavLink>
            <NavLink to='/favorites' className='nav'>Favorites</NavLink>
            <NavLink to='/motivation' className='nav'>Motivation</NavLink>
            <NavLink to='/positive' className='nav'>Positive</NavLink>
            <NavLink to='/life' className='nav'>Life</NavLink>
            <NavLink to='/funny' className='nav'>Funny</NavLink>
            <NavLink to='/love' className='nav'>Love</NavLink>
            <NavLink to='/random' className='nav'>Random</NavLink>
            <NavLink to='/students' className='nav'>Students</NavLink>
            <NavLink to='/inspiration' className='nav'>Inspiration</NavLink>
            <NavLink to='/meditation' className='nav'>Meditation</NavLink>
          </header>
        </div>
      </div>
    );
  }
}

export default Header;