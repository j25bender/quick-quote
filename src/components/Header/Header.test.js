import React from 'react';
import Header from './Header';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Header Component Test', () => {

  it('should render without blowing up (oh, and match snapshot)', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});