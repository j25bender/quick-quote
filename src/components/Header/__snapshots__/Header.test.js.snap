// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test should render without blowing up (oh, and match snapshot) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header-section"
>
        <header>
                <h1>
                        QUICK QUOTE
                </h1>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/home"
                >
                        Home
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/motivation"
                >
                        Motivation
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/positive"
                >
                        Positive
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/life"
                >
                        Life
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/funny"
                >
                        Funny
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/love"
                >
                        Love
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/random"
                >
                        Random
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/students"
                >
                        Students
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/inspiration"
                >
                        Inspiration
                </NavLink>
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/meditation"
                >
                        Meditation
                </NavLink>
        </header>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header>
          <h1>
                    QUICK QUOTE
          </h1>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/home"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/motivation"
          >
                    Motivation
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/positive"
          >
                    Positive
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/life"
          >
                    Life
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/funny"
          >
                    Funny
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/love"
          >
                    Love
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/random"
          >
                    Random
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/students"
          >
                    Students
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/inspiration"
          >
                    Inspiration
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav"
                    to="/meditation"
          >
                    Meditation
          </NavLink>
</header>,
        "className": "header-section",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              QUICK QUOTE
</h1>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/home"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/motivation"
>
              Motivation
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/positive"
>
              Positive
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/life"
>
              Life
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/funny"
>
              Funny
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/love"
>
              Love
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/random"
>
              Random
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/students"
>
              Students
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/inspiration"
>
              Inspiration
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/meditation"
>
              Meditation
</NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "QUICK QUOTE",
            },
            "ref": null,
            "rendered": "QUICK QUOTE",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav",
              "to": "/home",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Motivation",
              "className": "nav",
              "to": "/motivation",
            },
            "ref": null,
            "rendered": "Motivation",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Positive",
              "className": "nav",
              "to": "/positive",
            },
            "ref": null,
            "rendered": "Positive",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Life",
              "className": "nav",
              "to": "/life",
            },
            "ref": null,
            "rendered": "Life",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Funny",
              "className": "nav",
              "to": "/funny",
            },
            "ref": null,
            "rendered": "Funny",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Love",
              "className": "nav",
              "to": "/love",
            },
            "ref": null,
            "rendered": "Love",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Random",
              "className": "nav",
              "to": "/random",
            },
            "ref": null,
            "rendered": "Random",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Students",
              "className": "nav",
              "to": "/students",
            },
            "ref": null,
            "rendered": "Students",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Inspiration",
              "className": "nav",
              "to": "/inspiration",
            },
            "ref": null,
            "rendered": "Inspiration",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Meditation",
              "className": "nav",
              "to": "/meditation",
            },
            "ref": null,
            "rendered": "Meditation",
            "type": [Function],
          },
        ],
        "type": "header",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-section"
>
          <header>
                    <h1>
                              QUICK QUOTE
                    </h1>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/home"
                    >
                              Home
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/motivation"
                    >
                              Motivation
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/positive"
                    >
                              Positive
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/life"
                    >
                              Life
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/funny"
                    >
                              Funny
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/love"
                    >
                              Love
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/random"
                    >
                              Random
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/students"
                    >
                              Students
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/inspiration"
                    >
                              Inspiration
                    </NavLink>
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="nav"
                              to="/meditation"
                    >
                              Meditation
                    </NavLink>
          </header>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <header>
            <h1>
                        QUICK QUOTE
            </h1>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/home"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/motivation"
            >
                        Motivation
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/positive"
            >
                        Positive
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/life"
            >
                        Life
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/funny"
            >
                        Funny
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/love"
            >
                        Love
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/random"
            >
                        Random
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/students"
            >
                        Students
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/inspiration"
            >
                        Inspiration
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav"
                        to="/meditation"
            >
                        Meditation
            </NavLink>
</header>,
          "className": "header-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                QUICK QUOTE
</h1>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/home"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/motivation"
>
                Motivation
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/positive"
>
                Positive
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/life"
>
                Life
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/funny"
>
                Funny
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/love"
>
                Love
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/random"
>
                Random
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/students"
>
                Students
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/inspiration"
>
                Inspiration
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav"
                to="/meditation"
>
                Meditation
</NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "QUICK QUOTE",
              },
              "ref": null,
              "rendered": "QUICK QUOTE",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav",
                "to": "/home",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Motivation",
                "className": "nav",
                "to": "/motivation",
              },
              "ref": null,
              "rendered": "Motivation",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Positive",
                "className": "nav",
                "to": "/positive",
              },
              "ref": null,
              "rendered": "Positive",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Life",
                "className": "nav",
                "to": "/life",
              },
              "ref": null,
              "rendered": "Life",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Funny",
                "className": "nav",
                "to": "/funny",
              },
              "ref": null,
              "rendered": "Funny",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Love",
                "className": "nav",
                "to": "/love",
              },
              "ref": null,
              "rendered": "Love",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Random",
                "className": "nav",
                "to": "/random",
              },
              "ref": null,
              "rendered": "Random",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Students",
                "className": "nav",
                "to": "/students",
              },
              "ref": null,
              "rendered": "Students",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Inspiration",
                "className": "nav",
                "to": "/inspiration",
              },
              "ref": null,
              "rendered": "Inspiration",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Meditation",
                "className": "nav",
                "to": "/meditation",
              },
              "ref": null,
              "rendered": "Meditation",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
